src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -175,32 +183,43 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
       exportAllSelectedOptions: {},
       isIncludeReferencesDeepChecked: true,
       workspaceId: this.props.workspaces.client.currentWorkspaceId$.getValue(),
+      availableWorkspace: this.props.workspaces.client.workspaceList$.getValue(),
     };
   }
 
   private get workspaceIdQuery() {
-    return this.state.workspaceId ? ['public', this.state.workspaceId] : ['public'];
+    const { availableWorkspace } = this.state;
+    return availableWorkspace?.map((ws) => ws.id) ?? [GLOBAL_WORKSPACE_ID];
+    // return this.state.workspaceId ? ['public', this.state.workspaceId] : ['public'];
+  }
+
+  private get wsNameIdLookup() {
+    const { availableWorkspace } = this.state;
+    return availableWorkspace?.reduce((map, ws) => {
+      return map.set(ws.name, ws.id);
```

Comment:
Good question, that's my assumption. If there have two or more workspace with same name, it may hard for user to differentiate which one is which one.